src/components/modules/blockEvents.ts

```diff
@@ -332,17 +336,84 @@ export default class BlockEvents extends Module {
       /**
        * Merge Blocks
        */
-      this.mergeBlocks();
+      this.mergeBlocks(BlockManager.previousBlock, BlockManager.currentBlock);
+    }
+  }
+
+  /**
+   * Handle delete keydown on Block
+   *
+   * @param {KeyboardEvent} event - keydown
+   */
+  private deletePressed(event: KeyboardEvent): void {
+    const { BlockManager, BlockSelection, Caret } = this.Editor;
+    const currentBlock = BlockManager.currentBlock;
+    const tool = currentBlock.tool;
+
+    /**
+     * Check if Block should be removed by current Delete keydown
+     */
+    if (currentBlock.selected || (currentBlock.isEmpty && currentBlock.currentInput === currentBlock.firstInput)) {
+      event.preventDefault();
+
+      const index = BlockManager.currentBlockIndex;
+
+      if (BlockManager.nextBlock && BlockManager.nextBlock.inputs.length === 0) {
+        /** If previous block doesn't contain inputs, remove it */
+        BlockManager.removeBlock(index + 1);
+      } else {
+        /** If block is empty, just remove it */
+        BlockManager.removeBlock();
```

Comment:
See comment above, I don't think delete key should remove current block in any case

Reply:
As a windows user, I'm using the delete key more often for such tasks than the backspace key.
If you are selecting a block, pressing delete should delete it ( isn't that the point of the delete key, after all ? 😄 )

I have to admit though, that I copied that comment without understanding what it means. I don't see how it could be empty.